blueocean-core-js/src/js/components/DisablePipelineButton.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { Component, PropTypes } from 'react';
+import { Icon } from '@jenkins-cd/design-language';
+import { DisableJobApi as disableJobApi } from '../';
+import { stopProp } from '../utils';
+import { i18nTranslator } from '../i18n/i18n';
+
+const translate = i18nTranslator('blueocean-web');
+
+/**
+ * Disable a pipeline
+ */
+export class DisablePipelineButton extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            disabled: this.props.pipeline.disabled,
+            submitingChange: false,
+        };
+    }
+
+    _updateState(newDisabledState) {
+        this.setState({
+            disabled: newDisabledState,
+        });
+    }
+
+    _onDisableClick() {
+        disableJobApi.disable(this.props.pipeline).then(() => {
+            this._updateState(true);
+            this.props.onChangeDisableState(true);
+            this.setState({ submitingChange: false });
+        });
+    }
+
+    _onEnableClick() {
+        disableJobApi.enable(this.props.pipeline).then(() => {
+            this._updateState(false);
+            this.props.onChangeDisableState(false);
+            this.setState({ submitingChange: false });
+        });
+    }
+
+    render() {
+        const outerClass = this.props.className ? this.props.className : '';
```

Comment:
These two don't seem to be used.

Reply:
yeah I kept the same way of thinking as the run button, its not really necessary for the current use case, I'll delete it and if we will need it in the future we can add it back